tests/backbeat/tests/retry/retry.js

```diff
@@ -0,0 +1,164 @@
+const assert = require('assert');
+const crypto = require('crypto');
+const { series, parallel, times, timesSeries, doWhilst } = require('async');
+
+const { scalityS3Client, awsS3Client } = require('../../s3SDK');
+const ReplicationUtility = require('../../ReplicationUtility');
+const { makeGETRequest, makePOSTRequest, getResponseBody } =
+    require('../../utils/request');
+
+const scalityUtils = new ReplicationUtility(scalityS3Client);
+const awsUtils = new ReplicationUtility(awsS3Client);
+const srcBucket = `source-bucket-${Date.now()}`;
+const destFailBucket = process.env.AWS_S3_FAIL_BACKBEAT_BUCKET_NAME;
+const destFailLocation = process.env.AWS_S3_FAIL_BACKEND_DESTINATION_LOCATION;
+
+const hex = crypto.createHash('md5')
+    .update(Math.random().toString())
+    .digest('hex');
+const keyPrefix = `${srcBucket}/${hex}`;
+const key = `object-to-replicate-${Date.now()}`;
+const REPLICATION_TIMEOUT = 1200000;
+
+function checkMetrics(prevBacklog, prevCompletions, prevFailures, body) {
+    const { backlog, completions, failures } = body;
+    assert.strictEqual((backlog.results.count - prevBacklog.count), 0);
+    assert.strictEqual((backlog.results.size - prevBacklog.size), 0);
+    assert.strictEqual((completions.results.count - prevCompletions.count), 1);
+    assert.strictEqual((completions.results.size - prevCompletions.size), 1);
+    assert.strictEqual((failures.results.count - prevFailures.count), 0);
+    assert.strictEqual((failures.results.size - prevFailures.size), 0);
+}
+
+function performRetries(N, keys, done) {
```

Comment:
minor: I think you could get rid of parameter `N` and use `keys.length` instead

Reply:
Nice! Updated to remove that param.